plugins/outputs/signalfx/signalfx.go

```diff
@@ -383,22 +219,43 @@ func (s *SignalFx) GetObjects(metrics []telegraf.Metric, dps chan *datapoint.Dat
 		var metricType datapoint.MetricType
 		var metricTypeString string
 
-		metricType, metricTypeString = parseMetricType(metric)
+		// patch memory metrics to be gauge
+		if metric.Type() == telegraf.Counter && metric.Name() == "mem" {
+			metricType = datapoint.Gauge
+			metricTypeString = "gauge"
+		} else {
+			metricType, metricTypeString = GetMetricType(metric.Type())
+		}
 
-		for field := range metric.Fields() {
-			var metricName string
-			var metricProps = make(map[string]interface{})
+		for field, val := range metric.Fields() {
+			// Copy the metric tags because they are meant to be treated as
```

Comment:
That only copies the view object into the underlying map and will do basically nothing for immutability.  See https://play.golang.org/p/y0z57MTh7H-

Reply:
The only way to copy a map is to iterate each key and put it into a new map.

Reply:
Yes, that's what the Tags function does https://github.com/signalfx/telegraf/blob/signalfx-integration/metric/metric.go#L73